test/e2e/common/setup-kubestellar.sh

```diff
@@ -110,18 +109,16 @@ if [ "$use_release" = true ] ; then
   helm upgrade --install ks-core oci://ghcr.io/kubestellar/kubestellar/core-chart \
     --version $(yq .KUBESTELLAR_VERSION core-chart/values.yaml) \
     --kube-context $HOSTING_CONTEXT \
-    --set kubeflex-operator.install=false \
     --set-json='ITSes=[{"name":"its1"}]' \
     --set-json='WDSes=[{"name":"wds1"}]' \
     --set verbosity.kubestellar=${KUBESTELLAR_CONTROLLER_MANAGER_VERBOSITY} \
     --set verbosity.transport=${TRANSPORT_CONTROLLER_VERBOSITY}
 else
   make kind-load-image
+  helm dependency update core-chart/
```

Comment:
I first tested without this line, expecting the `--dependency-update` flag to get the job done. But it did not. I had the 0.6.2 kubeflex chart in core-chart/charts, and it did not get replaced with 0.6.3 as it should. Note that the doc at https://helm.sh/docs/helm/helm_upgrade/ for that flag says that it updates dependencies **IF THEY ARE MISSING** (which I find a bit ambiguous, but experimentation resolved it). I tested with Helm version 3.16.1.

Reply:
@MikeSpreitzer I had the same experience myself. The flag does not seem to check the chart version. I'm not sure if it is a bug or the desired behavior. You can add a command, at least in the test script, to remove charts from `charts/`.
Overall I think this is the least evil.